fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonConnector.java

```diff
@@ -76,20 +76,30 @@ public PaimonConnector(ConnectorContext context) {
         }
         paimonOptions.setString(WAREHOUSE.key(), warehousePath);
 
-        Configuration storageConfig = new Configuration();
-        cloudConfiguration.applyToConfiguration(storageConfig);
-        Iterator<Entry<String, String>> propsIterator = storageConfig.iterator();
-        while (propsIterator.hasNext()) {
-            Entry<String, String> item = propsIterator.next();
-            if (item.getKey().startsWith("fs.s3")) {
-                paimonOptions.setString(item.getKey(), item.getValue());
+        if (this.cloudConfiguration.getCloudType() == CloudType.AWS && this.cloudConfiguration instanceof AWSCloudConfiguration) {
+            AWSCloudConfiguration awsCloudConfiguration = (AWSCloudConfiguration) this.cloudConfiguration;
+            paimonOptions.set("s3.connection.ssl.enabled", String.valueOf(awsCloudConfiguration.getEnableSSL()));
+            paimonOptions.set("s3.path.style.access", String.valueOf(awsCloudConfiguration.getEnablePathStyleAccess()));
+            AWSCloudCredential awsCloudCredential = awsCloudConfiguration.getAWSCloudCredential();
+            if (!awsCloudCredential.getEndpoint().isEmpty()) {
+                paimonOptions.set("s3.endpoint", awsCloudCredential.getEndpoint());
+            }
+            if (!awsCloudCredential.getAccessKey().isEmpty()) {
+                paimonOptions.set("s3.access-key", awsCloudCredential.getAccessKey());
+            }
+            if (!awsCloudCredential.getSecretKey().isEmpty()) {
+                paimonOptions.set("s3.secret-key", awsCloudCredential.getSecretKey());
             }
```

Comment:
I have reached an error `Caused by: org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool.` when querying from a paimon table which has large amount of file, So I wonder if it is possible to add `fs.s3a.connection.maximum` option in `paimonOptions` for pushing down to paimon? 
Or maybe there is another way to avoid this error?